source/includes/fact-system-vars.rst

```diff
@@ -26,25 +28,144 @@
+
+       - ``log``: Query metrics are written to the ``mongosqld``
+         :doc:`log </reference/log-messages/>`.
+
+       - ``stitch``: Query metrics are written to a `Stitch project
+         <https://docs.mongodb.com/stitch/>`__.
+
+     - ``off``
+     - ``off``
+
+   * - ``mongodb_max_connection_size``
+     - integer
+     - The maximum size of memory in bytes that may be allocated for
+       evaluating any query on any given client connection. A value of
+       ``0`` specifies no limit.
+     - ``0``
+     - ``0``
+
+   * - ``mongodb_max_server_size``
+     - integer
+     - The maximum size of memory in bytes that may be allocated for
+       evaluating any query on |bi-short|. A value of ``0`` specifies
+       no limit.
+     - ``0``
+     - ``0``
+
+   * - ``mongodb_max_stage_size``
+     - integer
+     - The maximum size of memory in bytes that may be allocated for
+       evaluating any query in any given query evaluation stage. A value of
+       ``0`` specifies no limit.
+     - ``0``
+     - ``0``
+
    * - ``mongodb_max_varchar_length``
      - integer
      - Specifies the maximum string length returned for columns using
        the ``VARCHAR`` data type. A value of ``0`` specifies
        no limit.
      - ``0``
      - ``0``
+
+   * - ``mongodb_version_compatibility``
+     - string
+     - For mixed cluster MongoDB installations, the minimum version of
+       any process within the cluster.
+     - none
+     - none
+
+   * - ``mongodb_git_version``
+     - string
+     - The git version of MongoDB the |bi-short| is connected to for
+       a given client connection.
+     - none
+     - none
+
+   * - ``mongodb_topology``
+     - string
+     - The MongoDB cluster topology the |bi-short| is connected to for
+       a given client connection.
+     - none
+     - none
+
+   * - ``mongodb_version``
+     - string
+     - The MongoDB version the |bi-short| is connected to for a given 
+       client connection.
+     - none
+     - none
+
+   * - ``mongosqld_version``
+     - string
+     - The |bi-short| version.
+     - none
+     - none
+
+   * - ``optimize_cross_joins``
+     - boolean
+     - If enabled, cross joins are optimized into inner joins when possible.
+     - ``true``
+     - ``true``
+
+   * - ``optimize_evaluations``
+     - boolean
+     - If enabled, constant-folding is performed.
+     - ``true``
+     - ``true``
+
+   * - ``optimize_filtering``
+     - boolean
+     - If enabled, predicates in ``WHERE`` clauses are moved as close as possible
+       to the MongoDB data source they operate on.
+     - ``true``
+     - ``true``
+
+   * - ``optimize_inner_joins``
+     - boolean
+     - If enabled, inner joins are reordered for more optimal query execution.
+     - ``true``
+     - ``true``
+
+   * - ``optimize_self_joins``
+     - boolean
+     - If enabled, when any non-json mapping mode (such as ``schema_mapping_mode``)
+       is used, parent/progeny join queries are evaluated more optimally.
```

Comment:
This one is a bit tricky. Suggested description:

`If enabled, attempt to translate joins between tables with the same underlying MongoDB collection without $lookup`

Reply:
Done.